Source/MLXNN/PositionalEncoding.swift

```diff
@@ -167,3 +167,50 @@ final public class ALiBi: Module {
         return attentionScores + alibiMask
     }
 }
+
+public class SuScaledRotaryEmbedding: Module {
+    let dimensions: Int
+    let base: Float
+    let maxPositionEmbeddings: Int
+    let originalMaxPositionEmbeddings: Int
+    let scale: Float
+    let _freqs: MLXArray
+
+    public init(
+        dimensions: Int,
+        base: Float = 10000.0,
+        maxPositionEmbeddings: Int = 131072,
+        originalMaxPositionEmbeddings: Int = 4096,
+        // !! shortFactor is not used in the new Python implementation
```

Comment:
Indeed it's not. We may change that in the future, you can see my comment about it in [the relevant PR](https://github.com/ml-explore/mlx-examples/pull/945#issuecomment-2302479847). 

The basic problem is that switching frequencies during the generation is not correct. So you either have to go back and recompute the whole KV cache if you are doing a long generation or use a heuristic ahead of time to determine the length of the prompt + generation and use that to select the frequencies.